### PR TITLE
NEXUS-8818: Repository index mgt backend and task

### PR DESCRIPTION
**Pull request by cstamas**
Backend changes to expose Repository Index management functions.

Rebuilding assumes OOM-free browse methods:
https://github.com/sonatype/nexus-oss/pull/1590

Issue
https://issues.sonatype.org/browse/NEXUS-8818

CI
http://bamboo.s/browse/NX3-OSSF605

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java [hunk 1]
```diff
@@ -74,6 +75,34 @@ public SearchFacetImpl(final SearchService searchService,
...
+  @Override
+  @Guarded(by = STARTED)
+  public void rebuildIndex() {
+    log.info("Rebuilding index of repository {}", getRepository().getName());
+    rebuildIndexLock.writeLock().lock();
+    try {
+      searchService.deleteIndex(getRepository());
+      searchService.createIndex(getRepository());
+
+      // TODO: sort out with Stuart, is this correct, or would intermingle if this method is called from TX enabled method
```
Comment (sonatype-zion) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java [hunk 2]
```diff
@@ -74,6 +75,34 @@ public SearchFacetImpl(final SearchService searchService,
...
+      try (final StorageTx tx = getRepository().facet(StorageFacet.class).txSupplier().get()) {
+        tx.begin();
+        for (Component component : tx.browseComponents(tx.getBucket())) {
+          try {
```
Comment (sonatype-zion) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java]: MINOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 2 issues:
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by cstamas** [comment]
c5d55d4a0b9690667703d1cc3c6b3c0f56498836

**Pull request closed without merging**